src/partner-access/partner-access.service.ts

```diff
@@ -97,6 +98,52 @@ export class PartnerAccessService {
     });
   }
 
+  async getUserTherapySessions(): Promise<PartnerAccessEntity[]> {
+    return await this.partnerAccessRepository      
+      .createQueryBuilder('partnerAccess')
+      .leftJoin('partnerAccess.user', 'user') //get user associated with access code
+      .select([
+        'max(user.email) as userEmail', //get first user email. This will also return nulls as there is no way to add a condition to a joined column apparently
+        'sum(partnerAccess.therapySessionsRemaining) as therapyTotal', //get total therapy sessions available
+        'max(partnerAccess.accessCode) as partnerAccessCode', //get any access code - this will be used as an identifier to update. Uuids do not have aggregate functions in postgres so its a bit annoying to get that instead
+      ])
+      .where('partnerAccess.featureTherapy=true') //only get access codes with feature therapy turned on
+      .groupBy('partnerAccess.userId') //group by user as we can have users with multiple access codes
```

Comment:
@Zahra-8d for the tests to work - you need to mock the groupBy and and getRawMany in the Use Factory in line 74 of the test file. See if that makes it work